Content.Server/StationEvents/RampingStationEventSchedulerSystem.cs

```diff
@@ -16,30 +16,35 @@ public sealed class RampingStationEventSchedulerSystem : GameRuleSystem<RampingS
     [Dependency] private readonly EventManagerSystem _event = default!;
     [Dependency] private readonly GameTicker _gameTicker = default!;
 
-    public float GetChaosModifier(EntityUid uid, RampingStationEventSchedulerComponent component)
+    /// <summary>
+    ///     A <see href="https://www.desmos.com/calculator/87huunvoxq">logistic curve equation</see> used to smooth out the transition between event times at shift start, vs. shift end.
+    ///     Depending on the settings used, the end time might not necessarily be the point at which timers hit the floor.
+    ///     It is after all, an asymptote.
+    /// </summary>
+    /// <param name="component"></param>
+    /// <param name="startTime"></param>
+    /// <param name="endTimeOffset"></param>
+    /// <returns></returns>
+    public float RampingEventTimeEquation(RampingStationEventSchedulerComponent component, float startTime, float endTimeOffset = 0)
     {
-        var roundTime = (float) _gameTicker.RoundDuration().TotalSeconds;
-        if (roundTime > component.EndTime)
-            return component.MaxChaos;
-
-        return component.MaxChaos / component.EndTime * roundTime + component.StartingChaos;
+        var endTime = Math.Clamp(endTimeOffset, 0.1f, startTime - 1);
+        var shiftLength = Math.Max(1, _cfg.GetCVar(CCVars.EventsRampingAverageEndTime) - component.ShiftLengthOffset);
+        return 2 * endTime
+            / (1
+            	+ MathF.Exp(_cfg.GetCVar(CCVars.EventsRampingAverageChaos)
```

Comment:
0 reason to use Math.Exp here, this leads to exponential growth which is highly misleading